src/app/admin/layout.tsx

```
@@ -5,23 +5,42 @@ import { useAuth } from '@/contexts/auth-context'
   children: React.ReactNode
 }) {
-  const { user, loading, isAdmin } = useAuth()
+  const { user, loading } = useAuth()
+  const [isAdmin, setIsAdmin] = React.useState(false)
+  const [checkingAdmin, setCheckingAdmin] = React.useState(true)
   const router = useRouter()
 
   React.useEffect(() => {
-    if (!loading && (!user || !isAdmin)) {
+    async function checkAdminStatus() {
+      if (user && user.email) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)